plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -170,15 +170,15 @@ public void setPropertyValue(String path, Object value) {
     }
 
     public String getIndex() {
```

Comment:
I wonder if we should have a getMetadata(String name) method rather than all the distinct getters for each one of them. and maybe throw exception if the metadata is not there when we look for it.